### DISCUSSION TITLE
Is there a way to disable negative index loop?

### DISCUSSION THREAD

**Discussion by TheYellowArchitect** [godotengine/godot-proposals#5554] [Scripting]
Code-example:
```
int_list[-1] == int_list[int_list.size()]
```

I do not want this behaviour by default because when I code and an index is negative, I want the compiler to hit error, pause the game and highlight the line, so it's obvious a certain dynamic index is wrong. With the default negative loop index, errors like this pass unnoticed and are relatively hard to find within huge systems, where you will need at least an hour to find in a huge new system. 

If I want to make negative looping for indexes, I can make an array wrapper function myself, I don't want this counter-intuitive behaviour for every array[] type in existence. It is common to use -1 for error/null indexes for example, but this destroys that practice.

Anyway, is there an editor settings option or something i missed?

**Comment by dalexeev** [comment]
No, the only way is to check:

```gdscript
assert(0 <= i and i < array.size())
```

or

```gdscript
if i < 0 or i >= array.size():
    push_error("some error")
    return some_default
```

I also find this more error-prone than convenient. In my opinion, there should be a method (`get`?) for these purposes, and the `[]` syntax should be more strict. `String` in 4.0-dev works like this (~`s[-1]` is always an error~, `s.right(1)` returns the last character or an empty string).

**Reply by TheYellowArchitect** [reply]
I am glad to see I am not alone in this. I am also surprised that the ideal behaviour satisfying all users already exists for `String`
At least this problem is negligible, as it doesn't kill my workflow, so I can live with it
